src/Database/Connection.php

```diff
@@ -458,6 +511,41 @@ public function rollback()
     }
 
     /**
+     * Enables/disables the usage of savepoints, enables only if driver the allows it.
+     *
+     * If you are trying to enable this feature, make sure you check the return value of this
+     * function to verify it was enabled successfully.
+     *
+     * ### Example:
+     *
+     * `$connection->useSavePoints(true)` Returns true if drivers supports save points, false otherwise
+     * `$connection->useSavePoints(false)` Disables usage of savepoints and returns false
+     *
+     * @param bool $enable Whether or not save points should be used.
+     * @return $this
+     */
+    public function enableSavePoints($enable)
+    {
+        if ($enable === false) {
+            $this->_useSavePoints = false;
+        } else {
+            $this->_useSavePoints = $this->_driver->supportsSavePoints();
+        }
+
+        return $this;
+    }
+
+    /**
+     * Returns whether this connection is using savepoints for nested transactions
+     *
+     * @return bool true if enabled, false otherwise
+     */
+    public function isEnabledSavePoints()
```

Comment:
`getEnabledSavePoints()` is the first idea as you would expect the "save points" as integer or some kind of array to be returned. Completely kills the idea of a bool result.